### PR TITLE
Fork Ingress controller to support Ingress v1 API

### PR DESCRIPTION
**Pull request by howardjohn**
Supporting only Ingress/v1 means we lose support of Kubernetes 1.18
Supporting only Ingress/v1beta1 means we lose support of Kubernetes 1.22
Since supporting both in a monolith controller is painful due to lack of
usable conversion logic between
the two versions.
As a compromise, we instead just fork the controller. Once 1.18 support
is no longer needed, we can drop the old controller
For https://github.com/istio/istio/issues/30822

### CODE REVIEW

File: pilot/pkg/bootstrap/configcontroller.go [hunk 1]
```diff
@@ -83,22 +84,45 @@ func (s *Server) initConfigController(args *PilotArgs) error {
 	// If running in ingress mode (requires k8s), wrap the config controller.
 	if hasKubeRegistry(args.RegistryOptions.Registries) && meshConfig.IngressControllerMode != meshconfig.MeshConfig_OFF {
 		// Wrap the config controller with a cache.
-		s.ConfigStores = append(s.ConfigStores,
-			ingress.NewController(s.kubeClient, s.environment.Watcher, args.RegistryOptions.KubeOptions))
+		// Supporting only Ingress/v1 means we lose support of Kubernetes 1.18
+		// Supporting only Ingress/v1beta1 means we lose support of Kubernetes 1.22
+		// Since supporting both in a monolith controller is painful due to lack of usable conversion logic between
```
Comment (Monkeyanator) [inline pilot/pkg/bootstrap/configcontroller.go:89]:
```suggestion
		// Supporting both in a monolith controller is painful due to lack of usable conversion logic between
```

**Review by Monkeyanator — APPROVED** [review]
So we'll have to make sure changes in `/ingress` code happen in `/ingressv1` until `v1beta1` removal? Should we add a comment or maybe README to that effect? Or maybe it's enough to enforce through review process

**Comment by howardjohn** [comment]
> So we'll have to make sure changes in `/ingress` code happen in `/ingressv1` until `v1beta1` removal? Should we add a comment or maybe README to that effect? Or maybe it's enough to enforce through review process

Yep. The good news is we pretty much never change this. A readme or comment is probably good, I couldn't think of anywhere that would ensure it was read but not sure what else to do. The tests *should* fail in postsubmit/presubmit if they forget one of them, ideally

File: pilot/pkg/config/kube/ingressv1/controller.go [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+//   If we need more flexibility - we can add it (but likely we'll deprecate ingress support first)
+// -
+
+var schemas = collection.SchemasFor(
+	collections.IstioNetworkingV1Alpha3Virtualservices,
+	collections.IstioNetworkingV1Alpha3Gateways)
+
+// Control needs RBAC permissions to write to Pods.
+
+type controller struct {
```
Comment (nmittler) [inline pilot/pkg/config/kube/ingressv1/controller.go:76]: Is this all new code or was it pulled out of git history?

Comment (howardjohn) [reply pilot/pkg/config/kube/ingressv1/controller.go:76]: Essentially `cp -r ingress ingressv1; sed s/v1beta1/v1/g`. No major changes

**Review by nmittler — APPROVED** [review]

**Pull request merged**